### PR TITLE
LL-1483 Starring accounts functionality

### PR DESCRIPTION
**Pull request by juan-cortes**
Brings the starring functionality to LLD. This consists in several parts the main one being the sidebar where we can drag and drop starred accounts for easier access by our users. The star component is completely independent of the context and can be used wherever we want, in the same line of thought i've refactored the context menu which has allowed the **list/grid account items to no longer be connected** (maybe performance boost here).

Also added some eye-candy to the star action with a twitter-style animation when we star an account, this is done with a frame based animation in css following something similar to what's [described in this article](https://medium.com/@chrismabry/how-did-they-do-that-the-twitter-like-animation-2a473b658e43)

### We turn this
image

### Into this
star

### Screenshots
image

image
image

### Known issues

- ~Missing a final design for the Account header star placement~
- ~Missing a final design for the Account grid mode card star placement~

### Type

Feature

### Context

https://ledgerhq.atlassian.net/browse/LL-1483

### Parts of the app affected / Test plan

New feature in the sidebar. Test that starring works from context menu (list view), direct click in list mode, account header. Dragging to reorder in the main sidebar.

### CODE REVIEW

File: src/components/Bookmarks/index.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+}
+
+class Bookmarks extends PureComponent<{
+  bookmarkedAccounts: (Account | TokenAccount)[],
+  dragDropBookmarkAction: (*) => any,
+  toggleBookmarkAction: string => void,
+  location: Location,
+  push: string => void,
+}> {
+  BookmarkItem = (account: Account | TokenAccount, i) => {
```
[the remark this answers is not in the document]
Comment (juan-cortes) [reply src/components/Bookmarks/index.js]: indeed it should

File: src/components/Bookmarks/index.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  height: 60px;
+  flex: 1;
+  align-items: center;
+  display: flex;
+  padding: 0px 15px;
+  border-radius: 4px;
+  border: 1px solid transparent;
+  background: ${p => (p.active ? p.theme.colors.lightGrey : 'white')};
+  &:hover ${AccountName} {
+    color: ${p => p.theme.colors.dark}!important;
```
[the remark this answers is not in the document]
Comment (juan-cortes) [reply src/components/Bookmarks/index.js]: apparently not, when I initially had it without the component specific selector I did have to force it, this one worked without it, good catch

**Comment by dasilvarosa** [comment]
https://app.zeplin.io/project/5a672fa9c1729867100289cd/screen/5d39c9df421705777871d743

**Comment by juan-cortes** [comment]
Bookmarking gone, long live stars

**Comment by juan-cortes** [comment]
@gre @Arnaud97234 it's debatable whether to show a bookmarked empty token or not, both behaviours sort of make sense. User decided to bookmark, and user decided to hide. Which one takes priority?

**Comment by Arnaud97234** [comment]
> @gre @Arnaud97234 it's debatable whether to show a bookmarked empty token or not, both behaviours sort of make sense. User decided to bookmark, and user decided to hide. Which one takes priority?

Agree, @aramab will decide then

**Pull request merged**